src/ChemicalDiffusion/ChemicalDiffusion.jl

```diff
@@ -0,0 +1,312 @@
+
+function Transform_ChemicalDiffusion(name::F, CharDim::GeoUnits{U}) where {F, U<:Union{GEO,SI}}
+    Transform_ChemicalDiffusion(diffusion_database(name), CharDim)
+end
+
+function Transform_ChemicalDiffusion(p::AbstractChemicalDiffusion{T}, CharDim::GeoUnits{U}) where {T,U<:Union{GEO,SI}}
+    nondimensionalize(Transform_ChemicalDiffusion(p), CharDim)
+end
+
+function Transform_ChemicalDiffusion(pp::AbstractChemicalDiffusion{T}) where T
+
+    D0          = Value(pp.D0)
+    D0_2σ       = Value(pp.D0_2σ)
+    Ea          = Value(pp.Ea)
+    Ea_2σ       = Value(pp.Ea_2σ)
+    ΔV          = Value(pp.ΔV)
+    ΔV_2σ       = Value(pp.ΔV_2σ)
+    T_range_min = Value(pp.T_range_min)
+    T_range_max = Value(pp.T_range_max)
+    P0          = Value(pp.P0)
+
+    D0_SI          = uconvert(m^2 / s, D0 )
+    D0_2σ_SI       = uconvert(m^2 / s, D0_2σ)
+    Ea_SI          = uconvert(J / mol, pp.Ea)
+    Ea_2σ_SI       = uconvert(J / mol, pp.Ea_2σ)
+    ΔV_SI          = uconvert(m^3 / mol, pp.ΔV)
+    ΔV_2σ_SI       = uconvert(m^3 / mol, pp.ΔV_2σ)
+    T_range_min_SI = uconvert(K, T_range_min)
+    T_range_max_SI = uconvert(K, T_range_max)
+    P0_SI          = uconvert(Pa, P0)
+
+
+    args       = (Name=unsafe_string(pp.Name),
+                  Mineral=unsafe_string(pp.Mineral),
+                  Formula=unsafe_string(pp.Formula),
+                  Species=unsafe_string(pp.Species),
+                  D0=D0_SI, D0_2σ=D0_2σ_SI,
+                  Ea=Ea_SI, Ea_2σ=Ea_2σ_SI,
+                  ΔV=ΔV_SI, ΔV_2σ=ΔV_2σ_SI,
+                  T_range_min_SI=T_range_min_SI,
+                  T_range_max_SI=T_range_max_SI,
+                  P0=P0_SI,
+                  Orientation=unsafe_string(pp.Orientation),
+                  Crystallography=unsafe_string(pp.Crystallography),
+                  Buffer=unsafe_string(pp.Buffer),
+                  Fluid=unsafe_string(pp.Fluid)
+                  )
+
+    return DiffusionData(; args...)
+end
+
+function Transform_ChemicalDiffusion(pp::AbstractChemicalDiffusion{T}, kwargs::NamedTuple) where T
+
+    (; D0, D0_2σ, Ea, Ea_2σ, ΔV, ΔV_2σ, T_range_min, T_range_max, P0) = kwargs
+
+    D0_new          = isnothing(D0) ? Value(pp.D0) : Value(GeoUnit(D0))
+    D0_2σ_new       = isnothing(D0_2σ) ? Value(pp.D0_2σ) : Value(GeoUnit(D0_2σ))
+    Ea_new          = isnothing(Ea) ? Value(pp.Ea) : Value(GeoUnit(Ea))
+    Ea_2σ_new       = isnothing(Ea_2σ) ? Value(pp.Ea_2σ) : Value(GeoUnit(Ea_2σ))
+    ΔV_new          = isnothing(ΔV) ? Value(pp.ΔV) : Value(GeoUnit(ΔV))
+    ΔV_2σ_new       = isnothing(ΔV_2σ) ? Value(pp.ΔV_2σ) : Value(GeoUnit(ΔV_2σ))
+    T_range_min_new = isnothing(T_range_min) ? Value(pp.T_range_min) : Value(GeoUnit(T_range_min))
+    T_range_max_new = isnothing(T_range_max) ? Value(pp.T_range_max) : Value(GeoUnit(T_range_max))
+    P0_new          = isnothing(P0) ? Value(pp.P0) : Value(GeoUnit(P0))
```

Comment:
maybe you could simplify this by defining a couple of functions such as:

```julia
f(a,b) = Value(GeoUnit(a))
f(::Nothing,b) = Value(b)
```
then you can simply write e.g.:
```julia
D0_new = f(D0, pp.D0)
```
and so on